util/src/main/java/tc/oc/pgm/util/chunk/NullChunkGenerator.java

```diff
@@ -38,12 +38,17 @@ public ChunkData generateChunkData(
 
     ChunkData chunkData = super.createChunkData(world);
 
-    // For everyblock in the chunk set the biome to plains
+    // For every block in the chunk set the biome
     for (int x = 0; x < 16; x++) {
       for (int z = 0; z < 16; z++) {
-        biome.setBiome(x, z, Biome.PLAINS);
+        switch (world.getEnvironment()) {
+          case NETHER -> biome.setBiome(x, z, Biome.valueOf("NETHER_WASTES"));
+          case THE_END -> biome.setBiome(x, z, Biome.valueOf("THE_END"));
+          default -> biome.setBiome(x, z, Biome.PLAINS);
```

Comment:
The biome to use should be computed outside the loop to avoid it being looked up 256 times per generated chunk, as well as the unnecessary branching inside the loop